code/modules/spell_system/spells/spell_list/balefulshift.dm

```diff
@@ -0,0 +1,111 @@
+//basic transformation spell. Should work for most simple_animals
+
+/spell/targeted/shapeshift
+	name = "Shapeshift"
+	desc = "This spell transforms the target into something else for a short while."
+
+	school = "transmutation"
+
+	charge_type = Sp_RECHARGE
+	charge_max = 600
+
+	duration = 0 //set to 0 for permanent.
+
+	var/list/normal_transformations = list()
+	var/list/big_transformations = list()
+	var/list/newVars = list() //what the variables of the new created thing will be.
+
+	cast_sound = 'sound/weapons/emitter2.ogg'
+	var/revert_sound = 'sound/weapons/emitter.ogg' //the sound that plays when something gets turned back.
+	var/share_damage = 1 //do we want the damage we take from our new form to move onto our real one? (Only counts for finite duration)
+	var/drop_items = 1 //do we want to drop all our items when we transform?
+	var/list/protected_roles = list() //which roles are immune to the spell
+
+/mob/living
+	var/datum/weakref/polymorph_origin
+
+/proc/badpolymorph(mob/living/M, mob/user, share_damage = 1, drop_items = 0, revert_sound = 'sound/weapons/emitter.ogg', list/normal_transformations = list(), list/big_transformations = list(), duration = 0, list/newVars = list())
+	if(M.buckled)
+		M.buckled.unbuckle_mob()
+
+	if (M.polymorph_origin)
+		var/mob/living/origin = M.polymorph_origin.resolve()
+		if (!origin)
+			log_debug("shapeshift: target mob's origin no longer exists (deleted?), ignoring.")
+			return
+
+		origin.unshapeshift_from(M, revert_sound, share_damage)
+		return
+	var/new_mob = pick(prob(85) ? normal_transformations : big_transformations)
+	var/mob/living/trans = new new_mob(get_turf(M))
+	trans.polymorph_origin = WEAKREF(M)
+	for(var/varName in newVars) 
+		if(varName in trans.vars)
+			trans.vars[varName] = newVars[varName]
+
+	trans.name = "[trans.name]"
+	if(istype(M,/mob/living/carbon/human) && drop_items)
+		for(var/obj/item/I in M.contents)
+			if(istype(I,/obj/item/organ))
+				continue
+			M.drop_from_inventory(I)
+	if(M.mind)
+		M.mind.transfer_to(trans)
+	else
+		trans.key = M.key
+	var/atom/movable/overlay/effect = new /atom/movable/overlay(get_turf(M))
+	effect.density = 0
+	effect.anchored = 1
+	effect.icon = 'icons/effects/effects.dmi'
+	effect.layer = 3
```

Comment:
no magic numbers

Reply:
What?

Reply:
`layer = 3` is a magic number, we have defines for layers
`anchored` should also be `TRUE` and `density` should be `FALSE`